src/config/settings.js

```diff
@@ -62,6 +65,61 @@ function init() {
   return settings;
 }
 
+function safeFileTest(path, conditions) {
+  try {
+    fs.accessSync(path, conditions);
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+function fileExists(path) {
+  try {
+    fs.statSync(path);
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+function fileIsWritable(path) {
+  try {
+    fs.accessSync(path, fs.constants.W_OK);
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+function qq(fullpath) {
+  return /\s/.test(fullpath) ? `"${fullpath}"` : fullpath;
+}
+
+function parseSaveError(err) {
+  let msg = err.toString();
+  console.log(`settings save error: ${msg}`);
+  let p = new RegExp(`^Error:\\s*${err.code}:\\s*(.*?),\\s*${err.syscall}\\s+'?${err.path}`);
+  let m = p.exec(msg);
+  let friendlierMsg = `Error trying to ${err.syscall} ${err.path}`;
+  if (m) {
+    friendlierMsg += `: ${m[1]}`;
+  }
+  let parentPath = dirname(err.path);
+  if (err.code == 'EACCES') {
+    if (!fileExists(err.path)) {
```

Comment:
Hmm, that also fails if we couldn't read the parent (because `stat` would error out with something that isn't `ENOENT`).  But that's probably fine, since the important part is getting the user to look at the file.

Reply:
Not sure what the problem is here -- `fileExists` has a try/catch to deal with missing or unreadable paths.